include/mcl_3dl/state_6dof.h

```diff
@@ -0,0 +1,278 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef MCL_3DL_STATE_6DOF_H
+#define MCL_3DL_STATE_6DOF_H
+
+#include <algorithm>
+
+#include <mcl_3dl/pf.h>
+#include <mcl_3dl/quat.h>
+#include <mcl_3dl/vec3.h>
+
+namespace mcl_3dl
+{
+class State6DOF : public mcl_3dl::pf::ParticleBase<float>
+{
+public:
+  mcl_3dl::Vec3 pos;
+  mcl_3dl::Quat rot;
+  bool diff;
+  float noise_ll_;
+  float noise_la_;
+  float noise_al_;
+  float noise_aa_;
+  mcl_3dl::Vec3 odom_err_integ_lin;
+  mcl_3dl::Vec3 odom_err_integ_ang;
+  class RPYVec
+  {
+  public:
+    mcl_3dl::Vec3 v;
+    RPYVec()
+    {
+    }
+    explicit RPYVec(const mcl_3dl::Vec3 &v)
+    {
+      this->v = v;
+    }
+    RPYVec(const float &r, const float &p, const float y)
+    {
+      this->v.x = r;
+      this->v.y = p;
+      this->v.z = y;
+    }
+  };
+  RPYVec rpy;
+  float &operator[](const size_t i)override
+  {
+    switch (i)
+    {
+      case 0:
+        return pos.x;
+      case 1:
+        return pos.y;
+      case 2:
+        return pos.z;
+      case 3:
+        return rot.x;
+      case 4:
+        return rot.y;
+      case 5:
+        return rot.z;
+      case 6:
+        return rot.w;
+      case 7:
+        return odom_err_integ_lin.x;
+      case 8:
+        return odom_err_integ_lin.y;
+      case 9:
+        return odom_err_integ_lin.z;
+      case 10:
+        return odom_err_integ_ang.x;
+      case 11:
+        return odom_err_integ_ang.y;
+      case 12:
+        return odom_err_integ_ang.z;
+    }
+    return pos.x;
+  }
+  const float &operator[](const size_t i) const
+  {
+    switch (i)
+    {
+      case 0:
+        return pos.x;
+      case 1:
+        return pos.y;
+      case 2:
+        return pos.z;
+      case 3:
+        return rot.x;
+      case 4:
+        return rot.y;
+      case 5:
+        return rot.z;
+      case 6:
+        return rot.w;
+      case 7:
+        return odom_err_integ_lin.x;
+      case 8:
+        return odom_err_integ_lin.y;
+      case 9:
+        return odom_err_integ_lin.z;
+      case 10:
+        return odom_err_integ_ang.x;
+      case 11:
+        return odom_err_integ_ang.y;
+      case 12:
+        return odom_err_integ_ang.z;
+    }
+    return pos.x;
+  }
+  size_t size() const override
+  {
+    return 10;
```

Comment:
what kind of size is it?

Reply:
size allowed in `operator[]`

this must be 13 for current state model

Reply:
that makes sense to me if it's 13.